src/java/hep/lcio/implementation/sio/SIOTrack.java

```diff
@@ -78,6 +78,7 @@ class SIOTrack extends ITrack
       ndf = in.readInt() ;
       dEdx = in.readFloat();
       dEdxError = in.readFloat();
+      nholes = in.readInt() ;
```

Comment:
Similar version check as in c++ required. Something like this should work.
```suggestion
      if (SIOVersion.encode(major,minor) > SIOVersion.encode(2,21) {
        nholes = in.readInt() ;
      }
```